bin/oli/src/commands/bench/suite.rs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::commands::bench::report::{Report, SampleSet};
+use anyhow::{ensure, Context, Result};
+use opendal::Operator;
+use serde::Deserialize;
+use std::path::Path;
+use std::time::{Duration, Instant};
+
+#[derive(Deserialize, Debug)]
+struct BenchSuiteConfig {
+    /// Workload to run.
+    workload: Workload,
+
+    /// Number of parallel tasks to run.
+    ///
+    /// Default to 1.
+    parallelism: Option<u32>,
+
+    /// Size of file in bytes.
+    file_size: u32,
+
+    /// Maximum time to run the bench suite.
+    #[serde(with = "humantime_serde")]
+    timeout: Duration,
+}
+
+#[derive(Deserialize, Debug)]
+enum Workload {
+    #[serde(rename = "upload")]
+    Upload,
+    #[serde(rename = "download")]
+    Download,
+}
+
+pub struct BenchSuite {
+    config: BenchSuiteConfig,
+}
+
+impl BenchSuite {
+    pub fn load(path: &Path) -> Result<Self> {
+        let content = std::fs::read_to_string(path)?;
+        let config = toml::from_str::<BenchSuiteConfig>(&content)?;
+        ensure!(
+            config.file_size >= 4096,
+            "file_size must be greater or equal to 4096"
+        );
+        println!("Create bench suite with config: {config:?}");
+        Ok(BenchSuite { config })
+    }
+
+    pub fn run(self, op: Operator) -> Result<()> {
+        println!("Start running bench suite ...");
+        let start = Instant::now();
+
+        let timeout = self.config.timeout;
+        let parallelism = self.config.parallelism.unwrap_or(1);
+        let file_size = self.config.file_size;
+        let workload = match self.config.workload {
+            Workload::Upload => "upload".to_string(),
+            Workload::Download => "download".to_string(),
+        };
+
+        let rt = tokio::runtime::Builder::new_multi_thread()
```

Comment:
There is no other tasks while performing bench. Maybe we can just setup the global tokio runtime instead. So we don't need to play with tricks like `tokio::task::spawn_blocking`.